astroquery/utils/tap/taputils.py

```diff
@@ -217,26 +218,75 @@ def get_table_name(full_qualified_table_name):
     pos = full_qualified_table_name.rfind('.')
     if pos == -1:
         return full_qualified_table_name
-    name = full_qualified_table_name[pos+1:]
+    name = full_qualified_table_name[pos + 1:]
     return name
 
 
-def get_suitable_output_file(conn_handler, async_job, outputFile, headers,
-                             isError, output_format):
-    dateTime = datetime.now().strftime("%Y%m%d%H%M%S")
-    fileName = ""
-    if outputFile is None:
-        fileName = conn_handler.get_file_from_header(headers)
-        if fileName is None:
+def get_suitable_output_file(conn_handler, async_job, output_file, headers,
+                             is_error, output_format):
+    date_time = datetime.now().strftime("%Y%m%d%H%M%S")
+    if output_file is None:
+        file_name = conn_handler.get_file_from_header(headers)
+        if file_name is None:
             ext = conn_handler.get_suitable_extension(headers)
             if not async_job:
-                fileName = f"sync_{dateTime}{ext}"
+                file_name = f"sync_{date_time}{ext}"
             else:
                 ext = conn_handler.get_suitable_extension_by_format(
                     output_format)
-                fileName = f"async_{dateTime}{ext}"
+                file_name = f"async_{date_time}{ext}"
     else:
-        fileName = outputFile
-    if isError:
-        fileName += ".error"
-    return fileName
+        file_name = output_file
+    if is_error:
+        file_name += ".error"
+    return file_name
+
+
+def get_suitable_output_file_name_for_current_output_format(output_file, output_format):
+    """ renames the name given for the output_file if the results for current_output format are returned compressed by default
+    and the name selected by the user does not contain the correct extension.
+
+    output_file : str, optional, default None
+        file name selected by the user
+    output_format : str, optional, default 'votable'
+        results format. Available formats in TAP are: 'votable', 'votable_plain',
+        'fits', 'csv', 'ecsv' and 'json'. Default is 'votable'.
+        Returned results for formats 'votable' 'ecsv' and 'fits' are compressed
+        gzip files.
+
+    Returns
+    -------
+    A string with the new name for the file.
+    """
+    compressed_extension = ".gz"
+    format_with_results_compressed = ['votable', 'fits', 'ecsv']
+    output_file_with_extension = output_file
+
+    if output_file is not None:
+        if output_format in format_with_results_compressed:
+            # In this case we will have to take also into account the .fits format
+            if not output_file.endswith(compressed_extension):
+                warnings.warn('WARNING!!! By default, results in "votable" and "fits" format are returned in '
```

Comment:
Please issue a specific warning type here, so to help any potential catching for it. (E.g. `InputWarning`)

Reply:
Thanks for the suggestion. That would mean creating a custom warning. Why do you think that is better than using the default `UserWarning`?

Reply:
I think InputWarning is a bit more specific to describe this situation. Maybe just remove the `'WARNING!!!` part then?

Reply:
For now I have removed all of them. But I will convert it also into an InputWarning. Working now in increasing the coverage. However this will take me a bit of time since I will be a few days away. Have a nice weekend!